src/Cake.Common/Tools/MSBuild/MSBuildRunner.cs

```diff
@@ -150,7 +172,8 @@ protected override IEnumerable<FilePath> GetAlternativeToolPaths(MSBuildSettings
                 throw new ArgumentNullException("settings");
             }
 
-            var path = MSBuildResolver.GetMSBuildPath(_fileSystem, _environment, settings.ToolVersion, settings.PlatformTarget);
+            var path = MSBuildResolver.GetMSBuildPath(_fileSystem, _environment, settings.ToolVersion,
+                                                       settings.PlatformTarget.HasValue ? settings.PlatformTarget.Value : PlatformTarget.MSIL);
```

Comment:
Move this to it's own line for brevity like: `var platform = settings.PlatformTarget ?? PlatformTarget.MSIL`